src/main/java/hu/stepintomeetups/listeners/SkillCenterListener.java

```diff
@@ -0,0 +1,68 @@
+import lombok.RequiredArgsConstructor;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.javacord.api.entity.message.Message;
+import org.javacord.api.event.message.MessageCreateEvent;
+import org.javacord.api.listener.message.MessageCreateListener;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Listener which listens to the 'commands', 'list-skills', 'i-know' and 'i-dont-know' commands.
+ */
+@ApplicationScoped
+@DiscordMessageListener
+@RequiredArgsConstructor
+public class SkillCenterListener implements MessageCreateListener {
+
+    private static final String COMMANDS = "commands";
+    private static final String SKILLS_COMMAND = "list-skills";
+    private static final String I_KNOW_COMMAND = "i-know";
+    private static final String I_DONT_KNOW_COMMAND = "i-dont-know";
+
+    private final BotConfiguration botConfiguration;
+    private final RoleAssociationAction roleAssociationAction;
+    private final SkillsAction skillsAction;
+    private final AvailableCommandsAction availableCommandsAction;
+
+    @Override
+    public void onMessageCreate(MessageCreateEvent event) {
+        Message message = event.getMessage();
+        String content = message.getContent();
+        message.getUserAuthor()
+                .ifPresent(user -> {
+                    if (content.startsWith(getPrefixedCommand(COMMANDS))) {
+                        availableCommandsAction.sendAllCommandsToUser(user);
+                        message.addReaction(EmojiParser.parseToUnicode(":white_check_mark:"));
+                    } else if (content.startsWith(getPrefixedCommand(SKILLS_COMMAND))) {
+                        skillsAction.sendKnownSkillsToUser(user);
+                        message.addReaction(EmojiParser.parseToUnicode(":white_check_mark:"));
+                    } else if (content.startsWith(getPrefixedCommand(I_KNOW_COMMAND))) {
+                        String rawSkills = content.substring(getPrefixedCommand(I_KNOW_COMMAND).length());
+                        String[] skills = rawSkills.split(",");
+                        roleAssociationAction.addRolesToUser(user, Arrays.stream(skills).map(String::toLowerCase).map(String::trim).collect(Collectors.toList()));
+                        message.addReaction(EmojiParser.parseToUnicode(":white_check_mark:"));
+                    } else if (content.startsWith(getPrefixedCommand(I_DONT_KNOW_COMMAND))) {
+                        String rawSkills = content.substring(getPrefixedCommand(I_DONT_KNOW_COMMAND).length());
+                        String[] skills = rawSkills.split(",");
+                        roleAssociationAction.removeRolesFromUser(user, Arrays.stream(skills).map(String::toLowerCase).map(String::trim).collect(Collectors.toList()));
+                        message.addReaction(EmojiParser.parseToUnicode(":white_check_mark:"));
+                    }
```

Comment:
I know we don't have many commands here (and not sure if we will have more of them), but maybe it would worth to keep them separated and have their own components/classes. What do you think?

Reply:
Of course, this is a very raw implementation of it, I would like to rework it to use CDI based bean lookup, so this ugly if-else won't be there.